src/MCPClient/lib/clientScripts/verify_checksum.py

```diff
@@ -0,0 +1,276 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Archivematica.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import print_function, unicode_literals
+
+import datetime
+import os
+import subprocess
+import sys
+
+import django
+from django.db import transaction
+
+django.setup()
+from main.models import File
+
+from custom_handlers import get_script_logger
+from databaseFunctions import insertIntoEvents
+
+
+def concurrent_instances():
+    """Function docstring. N.B. Delete me."""
+    return 1
+
+
+logger = get_script_logger("archivematica.mcp.client.compare_hashes")
+
+
+class NoHashCommandAvailable(Exception):
+    """Provide feedback to the user if the checksum command cannot be found
+    for the provided checksum file.
+    """
+
+
+class PREMISFailure(Exception):
+    """Provide feedback to the user if there is a problem writing PREMIS event
+    information to the database.
+    """
+
+
+class Hashsum(object):
+    """Class to capture various functions around calling Hashsum as a mechanism
+    for comparing user-supplied checksums in Archivematica.
+    """
+
+    COMMANDS = {
+        "metadata/checksum.md5": "md5sum",
+        "metadata/checksum.sha1": "sha1sum",
+        "metadata/checksum.sha256": "sha256sum",
+        "metadata/checksum.sha512": "sha512sum",
+        "metadata/checksum.b2": "b2sum",
+    }
+
+    OKAY_STRING = ": OK"
+    FAIL_STRING = ": FAILED"
+    ZERO_STRING = "no properly formatted"
+    IMPROPER_STRING = "improperly formatted"
+    FAILED_OPEN = ": FAILED open or read"
+    EXIT_NON_ZERO = "returned non-zero exit status 1"
+
+    def __init__(self, path, job=print):
+        """Class constructor."""
+        try:
+            self.COMMAND = self.COMMANDS[path]
+            self.job = job
+            self.command_called = None
+        except KeyError:
+            raise NoHashCommandAvailable()
+
+    def _call(self, *args):
+        """Make the call to Python subprocess and record the command being
+        called.
+        """
+        self.command_called = (self.COMMAND,) + args
+        return self._decode(subprocess.check_output(self.command_called))
+
+    def count_and_compare_lines(self, hashfile, objectsdir):
+        """Count the number of lines in a checksum file and compare with the
+        number of objects being transferred. The requirement of hashsum as
+        this microservice job is written is that the mapping is 1:1. There
+        isn't space for an empty line at the end of the file.
+        """
+        lines = self._count_lines(hashfile)
+        objects = self._count_files(objectsdir)
+        if lines == objects:
+            return True
+        self.job.pyprint(
+            "{}: Comparison failed with {} checksum lines and {} "
+            "transfer files".format(self.get_ext(hashfile), lines, objects),
+            file=sys.stderr,
+        )
+        return False
+
+    def compare_hashes(self, hashfile, objectsdir):
```

Comment:
If I understand the code paths correctly, `hashfile` here is the same `path` the constructor received? If that's the case why is it necessary to pass it again?